core/src/main/java/bisq/core/api/CoreMoneroNodeService.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * This file is part of Haveno.
+ *
+ * Haveno is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Haveno is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Haveno. If not, see <http://www.gnu.org/licenses/>.
+ */
+package bisq.core.api;
+
+import bisq.core.user.Preferences;
+import bisq.core.xmr.MoneroNodeSettings;
+
+import bisq.common.config.Config;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import monero.common.MoneroRpcConnection;
+import monero.daemon.MoneroDaemonRpc;
+
+/**
+ * Manages a Monero node instance or connection to an instance.
+ */
+@Slf4j
+@Singleton
+public class CoreMoneroNodeService {
```

Comment:
This class should allow listeners to be added and notify when the node is started (with the daemon passed as an argument) or stopped. This will prompt the connections service to setConnection(daemon.getRpcConnection()) or checkConnection() so it becomes aware the current ndoe is offline and may automatically switch to the next best available connection depending on its settings.

Reply:
Restored the listener and call the correct functions.